scvi/external/sysvi/_model.py

```diff
@@ -0,0 +1,374 @@
+        categorical_covariate_embed_keys
+            Name of obs column with additional categorical covariate information. Embedding will be learned.
+            This can be useful if the number of categories is very large, which would increase memory usage.
+            If using this type of covariate representation please also cite
+            `scPoli <[https://doi.org/10.1038/s41592-023-02035-2]>`_ .
+        continuous_covariate_keys
+            Name of obs column with additional continuous covariate information.
+        covariate_categ_orders
+            Covariate encoding information. Should be used if a new adata is to be set up according
+            to setup of an existing adata. Access via adata.uns['covariate_categ_orders'] of already setup adata.
+        covariate_key_orders
+            Covariate encoding information. Should be used if a new adata is to be set up according
+            to setup of an existing adata. Access via adata.uns['covariate_key_orders'] of already setup adata.
+        system_order
+            Same as covariate_orders, but for system. Access via adata.uns['system_order']
+        """
+        setup_method_args = cls._get_setup_method_args(**locals())
+
+        if adata.shape[1] != len(set(adata.var_names)):
+            raise ValueError("Adata var_names are not unique")
+
+        # The used layer argument
+        # This could be also done via registry, but that is too cumbersome
+        adata.uns["layer_information"] = {
+            "layer": layer,
+            "var_names": list(adata.var_names),
+        }
+
+        # If setup is to be prepared wtr another adata specs make sure all relevant info is present
+        if covariate_categ_orders or covariate_key_orders or system_order:
+            assert system_order is not None
+            if (
+                categorical_covariate_keys is not None
+                or categorical_covariate_embed_keys is not None
+                or continuous_covariate_keys is not None
+            ):
+                assert covariate_categ_orders is not None
+                assert covariate_key_orders is not None
+
+        # Make system embedding with specific category order
+
+        # Define order of system categories
+        if system_order is None:
+            system_order = sorted(adata.obs[batch_key].unique())
+        # Validate that the provided system_order matches the categories in adata.obs[system_key]
+        if set(system_order) != set(adata.obs[batch_key].unique()):
+            raise ValueError(
+                "Provided system_order does not match the categories in adata.obs[system_key]"
+            )
+
+        # Make one-hot embedding with specified order
+        systems_dict = dict(zip(system_order, ([float(i) for i in range(0, len(system_order))])))
+        adata.uns["system_order"] = system_order
+        system_cat = pd.Series(
```

Comment:
We should remove these parts. They are handled by the registry and by setting up query data for new AnnData objects.